docs/api/components.md

```diff
@@ -33,6 +36,7 @@ import { appRoutes } from './routing';
 | ----------------- | ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
 | `routes`          | `Routes[]`                | Your application's routes                                                                                                                     |
 | `history`         | `History`                 | The history instance for the router, if omitted memory history will be used (optional but recommended)                                        |
+| `plugins`         | `Plugin[]`                | Plugin allows you to hook into Router API and extra login on route load/prefetch/etc                                       |
```

Comment:
what do you guys think, should we 
1) keep `plugins` optional and default to empty array `[]`. Side effect - it might be unclear that you need to now pass `resourcesPlugin` to the Router when upgrading to the new RRR version.
2) make `plugins` prop required

Reply:
I would go with option 1 as Router can be used without plugins. When upgrading with option 2 it will break typing and it might also null error for those not using types.

> it might be unclear that you need to now pass resourcesPlugin

Docs and release log should help with this right? Or maybe default to `[resourcesPlugin]`?

Reply:
> Docs and release log should help with this right?

In theory, if folks read it at all 

> Or maybe default to [resourcesPlugin]?
That's the exact behavior I'm trying to remove, in order to fully decouple `resources-plugin` from the Router

Reply:
@MonicaOlejniczak what do you think about it?

Reply:
I'd prefer option 2, make plugins prop required. It will make the breaking change / new API obvious and clear, forcing devs to check the release notes. As long as we document how to fix it in there too, it should require only few line changes (per Router use tho).
It will also open the door to remove unnecessary router props (resourceContext, resourceData), while if we keep the implicit creation we still need to allow those.

Reply:
I also prefer option 2 for the reasons Alberto mentioned

Reply:
sounds good, thanks for your feedback team